src/Math-Statistics/Collection.extension.st

```diff
@@ -0,0 +1,44 @@
+Extension { #name : #Collection }
+
+{ #category : #'*Math-Statistics' }
+Collection >> geometricMean [
+	"Answer with the geometric mean of the collection"
+	"#(1 1 2 3 4 5 5 6 6 7 8 9) geometricMean  >>> 3.8583980015011217"
+	"{ 4. 1.  1 / 32} geometricMean >>> 0.49999999999999994"
+	"#(3.14 1 4.56 0.333) geometricMean >>> 1.4776945822943937"
+	"{1/3. 2/3. 8/3. 16/3} geometricMean >>> 1.3333333333333335"
+
+	self detect: [ :i | i < 0 ] ifFound: [ Error new signal: 'The geometric mean should not be calculated on negative numbers' ] .
```

Comment:
maybe test `i <= 0`

Reply:
Thank you, I've changed the logic to be `<=0`